### PR TITLE
Switches out the three billion args of electrocute act for flags

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

As per title.
Also removes the override arg because it didn't do anything productive and nothing used it.
Also removes a bunch of shitcode from the shock anomaly.
Also renames 'safety' to 'SHOCK_NOGLOVES' so people actually know what it does.

### CODE REVIEW

**Comment by optimumtact** [comment]
based and flag pilled

**Comment by ninjanomnom** [comment]
oh neat someone was listening while I rambled in coding-general

File: code/__DEFINES/mobs.dm [hunk 1]
```diff
@@ -219,6 +219,10 @@
 
 #define MAX_CHICKENS 50
 
+#define SHOCK_NOGLOVES (1 << 0)
```
Comment (AnturK) [inline code/__DEFINES/mobs.dm]: Please describe what these mean now.

File: code/game/objects/effects/anomalies.dm [hunk 2]
```diff
@@ -160,18 +160,7 @@
...
-				M.electrocute_act(shockdamage, "[name]", safety=1)
-				return
-			M.electrocute_act(shockdamage, "[name]")
-			return
-		else
-			M.adjustFireLoss(shockdamage)
-			M.visible_message("<span class='danger'>[M] was shocked by \the [name]!</span>", \
-		"<span class='userdanger'>You feel a powerful shock coursing through your body!</span>", \
-		"<span class='hear'>You hear a heavy electrical crack.</span>")
+		M.electrocute_act(shockdamage, name, flags = SHOCK_NOGLOVES)
```
Comment (AnturK) [inline code/game/objects/effects/anomalies.dm]: I don't see you setting canshock here

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm [hunk 3]
```diff
@@ -1,124 +1,124 @@
...
+
+/mob/living/simple_animal/bot/secbot/ed209/Initialize(mapload)
+	. = ..()
+	set_weapon() //giving it the right projectile and firing sound.
+
+/mob/living/simple_animal/bot/secbot/ed209/bot_reset()
+	..()
+	set_weapon()
+
+/mob/living/simple_animal/bot/ed209/electrocute_act(shock_damage, source, siemens_coeff = 1, flags = NONE)
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/bot/ed209bot.dm]: Something went wrong with merge here i assume.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm (hunk 3 above)
Comment (PKPenguin321) [reply code/modules/mob/living/simple_animal/bot/ed209bot.dm]: So everybody knows, the intended behavior is that Beepsky and ED are shock immune, and Beepsky in fact gains a temporary speed boost from it

File: code/__DEFINES/mobs.dm (hunk 1 above)
Comment (optimumtact) [reply code/__DEFINES/mobs.dm]: Document them in the autodoc format pls

File: code/modules/mob/living/simple_animal/hostile/megafauna/swarmer.dm [hunk 4]
```diff
@@ -269,7 +269,7 @@ GLOBAL_LIST_INIT(AISwarmerCapsByType, list(/mob/living/simple_animal/hostile/swa
 		else
 			var/mob/living/L = target
 			L.attack_animal(src)
-			L.electrocute_act(10, src, safety = TRUE) //safety = TRUE means we don't check gloves... Ok?
+			L.electrocute_act(10, src, flags = SHOCK_NOGLOVES) //flags = SHOCK_SAFETY means we don't check gloves... Ok?
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/hostile/megafauna/swarmer.dm]: Could chop off that confusing comment now.

**Review by AnturK — DISMISSED** [review]

**Pull request merged**